### PR TITLE
Add session variable enable_hive_column_stats

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] others

## Which issues of this PR fixes ：

Fixes #
Fe will get hive columns stats when user query hive external table. If user's hms pressure is very high, fe will failed to get the hive column stats. Also affects the execution time of the query. Add a session variable here, the user skips acquire hive columns stats in cbo process.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 4 (25.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java [hunk 1]
```diff
@@ -298,7 +299,11 @@ public Void computeHMSTableScanNode(Operator node, ExpressionContext context, Ta
         List<ColumnRefOperator> requiredColumns = new ArrayList<>(colRefToColumnMetaMap.keySet());
 
         List<ColumnStatistic> columnStatisticList;
+
         try {
+            if (!optimizerContext.getSessionVariable().enableHiveColumnStats()) {
+                throw new Exception("Session variable " + SessionVariable.ENABLE_HIVE_COLUMN_STATS + " is false");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java:305]: Why throw an exception here？

Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java:305]: To goto following catch scope: use unknown column stats. Yes it's quite unusual, better to use if-condition stmt.

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -492,6 +494,13 @@
     @VariableMgr.VarAttr(name = SINGLE_NODE_EXEC_PLAN, flag = VariableMgr.INVISIBLE)
     private boolean singleNodeExecPlan = false;
 
+    @VariableMgr.VarAttr(name = ENABLE_HIVE_COLUMN_STATS, flag = VariableMgr.SESSION)
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:497]: flag = VariableMgr.SESSION is unnecessary.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java]: done

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 19 (05.26%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|17|00.00%|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|

**Review by dirtysalt — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**